runtime/caml/custom.h

```diff
@@ -32,6 +37,7 @@ struct custom_operations {
                     /*out*/ uintnat * bsize_64 /*size in bytes*/);
   uintnat (*deserialize)(void * dst);
   int (*compare_ext)(value v1, value v2);
+  const struct custom_fixed_length* fixed_length;
 };
```

Comment:
I wasn't expecting a pointer to `struct custom_fixed_length`, just a flat sub-structure.

Reply:
That was the original plan, but it got lost in a maze of incompatible compilers.

The issue is supporting `custom_fixed_length_default`, since the `custom_foo_default` macros are used both in initialiser lists and as rvalues in assignments. As a flat sub-structure, `custom_foo_default` could be:

  - `{0, 0}` - an initialiser list, not valid as an rvalue
  - `(struct custom_fixed_length){0, 0}` - a structure literal, valid C99 but rejected by MSVC
  - `empty_fixed_length` (some predefined `static const` object) - when used as an initialiser, deemed "non-constant" and rejected by C99 compilers (although valid C++, so MSVC might be OK with this)

Making it be a pointer and defining `custom_fixed_length_default` as `NULL` seemed the easiest way out. If you can think of a portable alternative, I'm happy to change this!

Reply:
You're right, there is no nice solution in C90.  A not very nice solution:
```
# define custom_fixed_length_default {0, 0}
```
for use in initializers (the most common use).  For assigning to a lvalue `e`, you could do
```
  { struct custom_fixed_length cfl = custom_fixed_length_default; e = cfl; }
```
Is it the case that `caml_finalize` is the only place where we need an assignment to an expression of type `struct custom_fixed_length`?  If yes there might be other ways to write this function.

Reply:
I think `caml_finalize` is indeed the only occurrence of `custom_fixed_length_default` in an assignment, and changing it isn't hard. I was more worried about external users of the API, who might be used to using `custom_foo_default` in both assignments and initialisers.